lms/djangoapps/grades/api/v1/tests/test_gradebook_views.py

```diff
@@ -647,6 +647,77 @@ def test_gradebook_data_for_single_learner(self, login_method):
                 actual_data = dict(resp.data)
                 self.assertEqual(expected_results, actual_data)
 
+    def test_gradebook_data_for_single_learner_override(self):
+        """
+        Tests that, when a subsection grade that was created from an override, and thus
+        does not have a truthy `first_attempted` attribute, is the only grade for a
+        user's subsection, we still get data including a non-zero possible score.
+        """
+        with patch('lms.djangoapps.grades.course_grade_factory.CourseGradeFactory.read') as mock_grade:
+            course_grade = self.mock_course_grade(self.student, passed=True, letter_grade='A', percent=0.85)
+
+            mock_subsection_grades = {
+                self.subsections[self.chapter_1.location][0].location: self.mock_subsection_grade(
+                    self.subsections[self.chapter_1.location][0],
+                    earned_all=1.0,
+                    possible_all=2.0,
+                    earned_graded=1.0,
+                    possible_graded=2.0,
+                    override=MagicMock(),
+                ),
+                self.subsections[self.chapter_1.location][1].location: self.mock_subsection_grade(
+                    self.subsections[self.chapter_1.location][1],
+                    earned_all=1.0,
+                    possible_all=2.0,
+                    earned_graded=1.0,
+                    possible_graded=2.0,
+                    override=MagicMock(),
+                ),
+                self.subsections[self.chapter_2.location][0].location: self.mock_subsection_grade(
+                    self.subsections[self.chapter_2.location][0],
+                    earned_all=1.0,
+                    possible_all=2.0,
+                    earned_graded=1.0,
+                    possible_graded=2.0,
+                    override=MagicMock(),
+                ),
+                self.subsections[self.chapter_2.location][1].location: self.mock_subsection_grade(
+                    self.subsections[self.chapter_2.location][1],
+                    earned_all=1.0,
+                    possible_all=2.0,
+                    earned_graded=1.0,
+                    possible_graded=2.0,
+                    override=MagicMock(),
+                ),
+            }
+            course_grade.subsection_grade = lambda key: mock_subsection_grades[key]
+            mock_grade.return_value = course_grade
+
+            with override_waffle_flag(self.waffle_flag, active=True):
+                self.login_course_admin()
```

Comment:
#NIT can you actually just do ddt and test for all global staff and course writable permissions?
See the next test below on how to do this.